test/optim/test_optim.py

```diff
@@ -807,14 +807,18 @@ def _test_foreach_memory(self, optimizer_pairs_with_flags):
             intermediate_size = nparams * param.nelement() * param.element_size()
             nintermediates = 1  # we expect a budget of 1 intermediate most of the time
             if (('capturable' in kwargs_with_flags and kwargs_with_flags['capturable']) or
-                    optimizer_constructor.__name__ == "Adadelta"):
+                    optimizer_constructor.__name__ in ["Adadelta", "ASGD"]):
                 # with capturable in Adam(W), we have 2 extra intermediates for the bias_corrections
                 # with Adadelta, we have 2 extra for (acc_delta + eps) and (square_avg + eps)
                 nintermediates = 3
                 if optimizer_constructor.__name__ == "NAdam":
                     # with capturable in NAdam, we have 3 extra intermediates for the
                     # bias_correction, mus, and mu_nexts
                     nintermediates = 5
+                elif optimizer_constructor.__name__ == "ASGD":
+                    # ASGD allocates axs, 2x mus, 2x etas, and grads at the same time
+                    nintermediates = 4
```

Comment:
With added dels below, we should be able to achieve only nintermediates = 2 for the capturable=False case, and i guess nintermediates = 4 is fine for the capturable. It's good to be as narrow as possible here.